src/js/Content/Features/Community/Badges/CBadges.js

```diff
@@ -15,20 +15,17 @@ export class CBadges extends CCommunityBase {
             FBadgeCalculations,
         ]);
 
-        this.hasMultiplePages = document.querySelector(".pagebtn") !== null;
+        this.myProfile = CommunityUtils.currentUserIsOwner();
+        this.hasMultiplePages = document.querySelector(".profile_paging") !== null;
     }
 
     // TODO Cache this somehow or apply both FBadgeDropsCount & FBadgeSortAndFilter at once when doing these requests
     async eachBadgePage(callback) {
         const baseUrl = `https://steamcommunity.com/${window.location.pathname}?p=`;
 
-        let skip = 1;
-        const m = window.location.search.match(/p=(\d+)/);
-        if (m) {
-            skip = parseInt(m[1]);
-        }
+        const skip = parseInt(new URL(window.location.href).searchParams.get("p")) || 1;
 
-        const lastPage = parseInt(DOMHelper.selectLastNode(document, ".pagelink").textContent);
+        const lastPage = parseInt(DOMHelper.selectLastNode(document, "a.pagelink").textContent);
```

Comment:
Why specify the tag here?

Reply:
I though this would make it clearer we're selecting one of those clickable page number links 1,2,3..., also
since we're searching the whole `document` here, it shouldn't hurt to be a bit more specific?